main.go

```diff
@@ -123,8 +123,8 @@ func main() {
 
 	flag.IntVar(&webhookPort,
 		"webhook-port",
-		9443,
-		"Webhook server port (set to 0 to disable)",
+		0,
+		"Webhook Server port, disabled by default. When enabled, the manager will only work as webhook server, no reconcilers are installed.",
 	)
```

Comment:
We need some logic in `main.go` to turn off the controllers when the webhook server is enabled, as per https://master.cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html

> The controller MUST register reconcilers if and only if webhookPort == 0.
> The controller MUST register webhooks if and only if webhookPort != 0.

Reply:
Thank you! Logic added here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1575/commits/6f3090142dd512f07fd315d1de508f3ffbc4eef9.

Reply:
Great!